### ISSUE TITLE
Signature Areas not defined -> Leading to Server Overload and Errors

### ISSUE THREAD

**Issue by blackerking** [LibreSign/libresign#4104]
Description

When initiating the signing process in LibreSign, it is possible to omit the signature area. So there are min. 1 signature requested but no place to sign.
Despite this omission, the signing process can still be started. This leads to significant issues, including:

The system attempts to handle the signing process even without a defined signature area, which starts the process and makes a heavy load on the server side. While this the UI is not responding and or makes 0byte files. The whole procedure seems to break.

Steps to Reproduce

    Open the LibreSign interface.
    Create a request for signing without defining a signature area.
    Start the signing process.

Expected Behavior

    The signing process should not be allowed to start if the signature areas is not defined
    The system should validate the presence of a signature area before proceeding to the signing process.
    Appropriate error messages should be displayed to inform the user about the missing signature area.
    In short: (requested_signature_fields_count == requested_signature_count)

Proposed Solution

    Implement a validation check to ensure a signature area is defined before allowing the signing process to begin.
    Provide clear error messages to the user if no signature area is set.
    Optimize the handling of such cases to prevent unnecessary server strain.

**Comment by vitormattos** [comment]
> When initiating the signing process in LibreSign, it is possible to omit the signature area. So there are min. 1 signature requested but no place to sign.

This is the expected behavior. Is possible to add a visible signature or not. When we digitally sign a PDF file, also is possible to only put the signature at document content, not visible by humans without a specific software to do this. This is other reason that is possible to put a qr-code at the footer of document to have a visible way to identify that the document was digitally signed.

> makes a heavy load on the server side.

Could you provide more details about this? Maybe this could be a specific behavior at your environment and would be good to have more details to identify if will be possible reproduce and if is a problem related to LibreSign and not about infrastructure, we can check a way to implement more improvements at this point.

> The system should validate the presence of a signature area before proceeding to the signing process.

This is a new feature that we can implement because now the visible signature isn't mandatory, but your description could be solved by adding a LibreSign Administration Setting to only allow to request to sign with a visible signature. Is a good idea, when we got a customer helping to fund this, we will be able to implement.

**Comment by blackerking** [comment]
We don't like to use the certificate function of libresign, just plain signatures inside documents.
Libresign doesn't check aganist this setting and the missing signature fields. So there is "nothing" to do.

**Comment by vitormattos** [comment]
There is a signature method called simple signature. In this method, you simply add an image to a PDF to say that it is signed without anything to ensure that the PDF has been changed after being signed. However, LibreSign does not work this way.

Signing with LibreSign implements the standard [PAdES](https://en.wikipedia.org/wiki/PAdES) and the PDF is cryptographically signed with the signing key of the person signing. This signing key can be created by LibreSign itself or created by a service provider in your country.

When the signing key used to sign in LibreSign is created by a service provider in your country, it is called a [qualified signature](https://en.wikipedia.org/wiki/Qualified_electronic_signature) within the eIDAS standard, for example [eIDAS](https://en.wikipedia.org/wiki/EIDAS). LibreSign accepts both types of signatures and both are [PAdES](https://en.wikipedia.org/wiki/PAdES) cryptographic standards.

To digitally sign a document, it is not necessary to insert visible fields and if you do not insert them, it is not mandatory to insert them. The digital signature is an internal cryptographic process of the document to ensure that after a signature the document has not been altered. Inserting fields with visible signature in the PDF is something only to improve the understanding of people who do not understand that the document was digitally signed.

Regardless of the way the signature is inserted in the document, the document after being signed by LibreSign can be cryptographically validated and if a bit in the PDF is changed, the signature is invalidated.

> Libresign doesn't check aganist this setting and the missing signature fields. So there is "nothing" to do.

When signing the document, if a visible signature field is placed, in all places where this field exists, only the person who is associated with this field will sign in that location. If several people sign, you can add several fields, or not, again, adding fields is optional and is not something that makes the signature simple, advanced or qualified because LibreSign works with the [PAdES](https://en.wikipedia.org/wiki/PAdES) standard.

**Comment by vitormattos** [comment]
Could you check again at the newest release of LibreSign?

**Comment by vitormattos** [comment]
Closing this issue as without interaction and feedback after many days.
